### ISSUE TITLE
Feature: `extension.icon`

### ISSUE THREAD

**Issue by Nazz78** [SketchUp/api-issue-tracker#3]
Add a method to SketchupExtension class which will allow us to define icon of the extension.

The icon is visible when user opens Extension Manager Window and it is currently visible only when extension is loaded from Extension Warehouse (or if we include extension_info.txt from it :-).

**Comment by thomthom** [comment]
Logged as SU-38303

**Comment by DanRathbun** [comment]
_<echoing my comments from topic #683>_

Currently the icon is being downloaded as a web resource from the EW. It requires that the developer manually upload the icon to their product page.

A method isn't really needed. If a common name were to be finalized and published, developers could just include an image in their RBZ. (Similar to how the RBZ archive contains a "**extension_info.txt**" file that is read by the API's [`SketchupExtension`](https://ruby.sketchup.com/SketchupExtension.html) class.)

> @thomthom  said (NOV 2019) in forum topic "[How do you get a custom icon in the Extension Manager](https://forums.sketchup.com/t/how-do-you-get-a-custom-icon-in-the-extension-manager/108953/3)", post 3:
>
> _We have been talking about adding an interface for extensions to bundle an icon with the extension and have Extension Manager use that._

In the spirit of the existing "**extension_info.txt**" file, I believe that I had proposed filenames using "extension_icon", ie (chosen in this order): "extension_icon.svg" (or "extension_icon.pdf" on Mac), "extension_icon.png", "extension_icon.jpg", "extension_icon.bmp" ... _etc. (Meaning that the [`Sketchup::ExtensionsManager`](https://ruby.sketchup.com/Sketchup/ExtensionsManager.html) chooses the **best** icon available depending upon the need such as display scaling, etc.)

When uploading to the Warehouse (for those extension that publish there) the inclusion of images with the "extension_icon" name **present in the extension subfolder** should take precedence over a manually uploaded icon.

----

But I am **not** opposed to the [`SketchupExtension`](https://ruby.sketchup.com/SketchupExtension.html) class getting a new **`icon`** property so that extensions privately published (or published elsewhere than the EW) can enjoy their icons displaying in the Extension Manager.

An `icon` property for extensions could allow their images to be placed in a nested subfolder rather than the extension's main subfolder **and / or** specify a custom filename. Ie ...
```ruby
ext.icon= File.join(__dir__,"icons/manager_icon")
```
The manager should know what file extension (image type) that is best to use.

**Comment by DanRathbun** [comment]
> @Fredosixx in https://github.com/SketchUp/api-issue-tracker/issues/683#issuecomment-894827895 said: _Anyway, if the issue #3, opened in October 2017, is still not resolved after almost 4 years, there is little hope than mine would be solved._

It is probably a matter of priority. This issue (I think) has been considered cosmetic.

But as we saw in the public forum recently, users see a grayed out icon (in the Extensions Manager) and think that this signifies something as not working or having not been loaded correctly. So this is also (I will argue) an UX usability / interface issue.

**Comment by Matherone** [comment]
Yes, enabling us developers to to set an extension icon independently of Extension Warehouse would be great. Preferred formats would include .svg and .png.

The current extension manager placeholder icon implies to end users that these extensions are:
a) Grey, similar, and technical
b) Probably incorrectly installed
c) Not trustable because you can see the developer missed something

In the Sketchup Studio package, Trimble now includes V-Ray, at a premium price. Yet the extension manager still brands V-Ray with another generic placeholder icon.

It's a tiny feature. 

But it would add value to a growing swathe of premium extensions distributed off the Extension warehouse. And make the extension experience more approachable. Especially for people trying out extensions. 

Icons help identify, individualise, and befriend extensions.

Trimble, would you please harvest this ripe and low-hanging fruit?
